src/cursor/cur_index.c

```diff
@@ -240,7 +240,17 @@ __curindex_search(WT_CURSOR *cursor)
 	found_key = child->key;
 	if (found_key.size < cursor->key.size)
 		WT_ERR(WT_NOTFOUND);
-	found_key.size = cursor->key.size;
+
+	/*
+	 * Custom collators will use unpack functions to
+	 * get the visible key fields. We need to provide
+	 * those fields in their entirety.
+	 */
+	if (cindex->index->collator != NULL)
+		WT_ERR(__wt_struct_repack(session, cindex->child->key_format,
+		    cindex->iface.key_format, &child->key, &found_key));
```

Comment:
Minor point: we've got `cindex->child` in local variable `child` (I think) -- can we use that consistently?

Can you please expand the comment a little and reformat?  Something like "Custom collators expect to see complete keys, pass an item containing all of the visible fields so it unpacks correctly".